src/lib/ng-v5/entry-point/resources/stylesheet-processor.ts

```diff
@@ -0,0 +1,111 @@
+import * as path from 'path';
+import * as log from '../../../util/log';
+import { CssUrl } from '../../../ng-package-format/shared';
+import { execSync } from 'child_process';
+
+// CSS Tools
+import * as autoprefixer from 'autoprefixer';
+import * as browserslist from 'browserslist';
+import * as nodeSassTildeImporter from 'node-sass-tilde-importer';
+import * as postcss from 'postcss';
+import * as postcssUrl from 'postcss-url';
+import postcssClean from './postcss-clean';
+import * as sass from 'node-sass';
+import * as stylus from 'stylus';
+
+export class StylesheetProcessor {
+  private postCssProcessor: postcss.Processor;
+
+  constructor(readonly basePath: string, readonly cssUrl?: CssUrl, readonly styleIncludePaths?: string[]) {
+    this.postCssProcessor = this.createPostCssProcessor(basePath, cssUrl, styleIncludePaths);
+  }
+
+  process(filePath: string, content: string) {
```

Comment:
Note: we made stylesheet processing sync, because it's triggered through the TypeScript transformations (which is a sync API).